### PR TITLE
Cluster inventory script

### PR DESCRIPTION
**Pull request by clayg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,15 +16,17 @@ controllers via the SwiftStack API
...
+   prefixing the command with the `docker run` command.
+4. The docker image exports a volume called `/output` which can be
+   mapped using the `-v` parameter (e.g. `-v `pwd`:/output`. This is
+   necessary if writing out files with the utility.
+5. If you want the container to use the hosts hostname resolution you
+   need to add --network=host
 
 ```
-$ docker run swiftstack/ssapi ss-util ...
+alias ss-util='docker run -v `pwd`:/output --network=host swiftstack/ssapi ss-util'
```
Comment (clayg) [inline README.md]: it'd be nice to merge #23

File: setup.py [hunk 2]
```diff
@@ -24,7 +24,8 @@
     author_email='trey@swiftstack.com',
     description='utilization gathering tools for SwiftStack API',
     entry_points={
-        'console_scripts': ['ss-util=swiftstackapi.cli:main']
+        'console_scripts': ['ss-util=swiftstackapi.cli:main',
+                            'ss-inv=swiftstackapi.inventory:main']
```
Comment (clayg) [inline setup.py]: This is probably the most important part to get consensus - what do we *call* this script!?

File: swiftstackapi/inventory.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        children[region_key]['children'][zone_key]['hosts'][
+            node['hostname']] = {
+                'cluster_facing_ip': node['cluster_facing_ip'],
+                'outward_facing_ip': node['outward_facing_ip'],
+                'replication_facing_ip': node['replication_facing_ip'],
+                'other_ips': node['other_ips']
+        }
+    data = {'all': {'children': {'swiftstacknodes': {
+        'children': children,
+    }}}}
```
Comment (clayg) [inline swiftstackapi/inventory.py]: like, this depth is sort of a mess :man_shrugging:

File: setup.py (hunk 2 above)
Comment (treyd) [reply setup.py]: i love `ss-inv` - i was going to call it something stupid like `ss-describe`

File: swiftstackapi/inventory.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                                     apiuser=args.ssapi_user,
+                                     apikey=args.ssapi_key,
+                                     controller_cert=args.controller_cert_path)
+    # XXX right now this is --cluster 0
+    if not args.cluster_id:
+        clusters = client.list_clusters()
+        logging.info('CLUSTERS:')
+        for cluster in clusters:
+            logging.info("  (%s) cluster: %r", cluster['id'], cluster)
+        logging.info('Choose cluster: ')
```
[the remark this answers is not in the document]
Comment (treyd) [reply swiftstackapi/inventory.py]: yeah, my opinion is the script shouldn't prompt for anything after the initial invocation

**Comment by treyd** [comment]
lgtm, go ahead and merge

**Pull request merged**